packages/location-management/src/components/EditLocationUnit/index.tsx

```diff
@@ -144,7 +142,44 @@ const EditLocationUnit = (props: EditLocationUnitProps) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [locId]);
 
-  if (loading) {
+  const locationUnits = useQuery(
+    LOCATION_UNIT_FIND_BY_PROPERTIES,
+    () => getBaseTreeNode(opensrpBaseURL),
+    {
+      onError: () => sendErrorNotification(lang.ERROR_OCCURRED),
+      select: (res: LocationUnit[]) => res,
+    }
+  );
+
+  useQueries(
+    locationUnits.data
+      ? locationUnits.data.map((location) => {
+          return {
+            queryKey: [LOCATION_HIERARCHY, location.id],
+            queryFn: () => new OpenSRPService(LOCATION_HIERARCHY, opensrpBaseURL).read(location.id),
+            onError: () => sendErrorNotification(lang.ERROR_OCCURRED),
+            // Todo : useQueries doesn't support select or types yet https://github.com/tannerlinsley/react-query/pull/1527
+            onSuccess: (res) => {
+              // the Tree is not already in the locationtree
+              if (!treeData.find((data) => JSON.stringify(data) === JSON.stringify(res))) {
+                // if the tree already exist i.e only a part of tree is changed, we need to replce old one else add new one
+                const alreadyExist = treeData.findIndex(
```

Comment:
I think it's simpler to check if the tree already exist (is an update) first, instead of checking if the whole object is in locationtree?

Reply:
https://github.com/opensrp/web/pull/645